xarray/core/parallelcompat.py

```diff
@@ -0,0 +1,199 @@
+"""
+The code in this module is an experiment in going from N=1 to N=2 parallel computing frameworks in xarray.
+It could later be used as the basis for a public interface allowing any N frameworks to interoperate with xarray,
+but for now it is just a private experiment.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import numpy as np
+
+from . import indexing, utils
+from .pycompat import DuckArrayModule, is_duck_dask_array
+
+CHUNK_MANAGERS = {}
+
+
+def _get_chunk_manager(name: str) -> "ChunkManager":
+    if name in CHUNK_MANAGERS:
+        chunkmanager = CHUNK_MANAGERS[name]
+        return chunkmanager()
+    else:
+        raise ImportError(f"ChunkManager {name} has not been defined")
+
+
+class ChunkManager(ABC):
+    """
+    Adapter between a particular parallel computing framework and xarray.
+
+    Attributes
+    ----------
+    array_type
+        Type of the array class this parallel computing framework provides.
+
+        Parallel frameworks need to provide an array class that supports the array API standard.
+        Used for type checking.
+    """
```

Comment:
@shoyer @dcherian @keewis this is the general approach I was thinking about using internally for now to abstract away the role of the parallel computing framework. Let me know if you have any comments on this approach.